### PR TITLE
feat(Sym2): Add lemmas for equivSym

### PR DESCRIPTION
**Pull request by Jun2M**
There is a `equivSym` defined in "Data/Sym/Sym2.lean" with no lemma about what its properties are. This PR attempts to change that.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Sym/Sym2.lean [hunk 1]
```diff
@@ -426,6 +426,12 @@ theorem IsDiag.mem_range_diag {z : Sym2 α} : IsDiag z → z ∈ Set.range (@dia
 theorem isDiag_iff_mem_range_diag (z : Sym2 α) : IsDiag z ↔ z ∈ Set.range (@diag α) :=
   ⟨IsDiag.mem_range_diag, fun ⟨i, hi⟩ => hi ▸ diag_isDiag i⟩
 
+@[simp]
+lemma map_IsDiag_iff (f : α ↪ β) (s : Sym2 α) :
```
Comment (urkud) [inline Mathlib/Data/Sym/Sym2.lean:430]: I think that it's better to use `f : α → β` with `(hf : f.Injective)`. Also, please use implicit args.

File: Mathlib/Data/Sym/Sym2.lean [hunk 2]
```diff
@@ -566,6 +572,49 @@ def sym2EquivSym' : Equiv (Sym2 α) (Sym' α 2) where
...
   Equiv.trans sym2EquivSym' symEquivSym'.symm
 
+@[simp]
+theorem equivSym_pair_eq {a b : α} : (equivSym α) s(a, b) = ⟨{a, b}, rfl⟩ := rfl
+
+@[simp]
+theorem equivSym_symm_eq_pair {a b : α} : (equivSym α).symm ⟨a ::ₘ {b}, rfl⟩ = s(a, b) := rfl
+
+@[simp]
+theorem mem_equivSym_iff_mem (s : Sym2 α) (a : α) : a ∈ equivSym α s ↔ a ∈ s := by
```
Comment (urkud) [inline Mathlib/Data/Sym/Sym2.lean:582]:
```suggestion
theorem mem_equivSym_iff_mem {s : Sym2 α} {a : α} : a ∈ equivSym α s ↔ a ∈ s := by
```

File: Mathlib/Data/Sym/Sym2.lean [hunk 3]
```diff
@@ -566,6 +572,48 @@ def sym2EquivSym' : Equiv (Sym2 α) (Sym' α 2) where
...
+  rw [Equiv.eq_symm_apply] at h
+  subst m
+  exact mem_equivSym_iff_mem.symm
+
+@[simp]
+theorem mem_equivSym_symm_iff_mem {m : Sym α 2} {a : α} : a ∈ (equivSym α).symm m ↔ a ∈ m :=
+  mem_equivSym_symm_iff_mem_of_eq rfl
+
+theorem equivSym_map_comm {f : α → β} {s : Sym2 α} :
+    (Sym2.equivSym β) (s.map f) = (Sym2.equivSym _ s).map f := by
```
Comment (vihdzp) [inline Mathlib/Data/Sym/Sym2.lean:600]:
```suggestion
    Sym2.equivSym β (s.map f) = (Sym2.equivSym _ s).map f := by
```

File: Mathlib/Data/Sym/Sym2.lean [hunk 4]
```diff
@@ -566,6 +572,48 @@ def sym2EquivSym' : Equiv (Sym2 α) (Sym' α 2) where
...
+  induction' s with x y
+  rfl
+
+theorem IsDiag.equivSym_symm_replicate (a : α) :
+    ((equivSym α).symm (Sym.replicate 2 a)).IsDiag  := by
```
Comment (vihdzp) [inline Mathlib/Data/Sym/Sym2.lean:605]:
```suggestion
    ((equivSym α).symm (Sym.replicate 2 a)).IsDiag := by
```

File: Mathlib/Data/Sym/Sym2.lean [hunk 5]
```diff
@@ -566,6 +572,48 @@ def sym2EquivSym' : Equiv (Sym2 α) (Sym' α 2) where
...
+  simp only [replicate, Multiset.replicate_succ, Multiset.replicate_zero, Multiset.cons_zero,
+    equivSym_symm_eq_pair, isDiag_iff_proj_eq]
+
+theorem equivSym_IsDiag_eq_replicate {s : Sym2 α} :
```
Comment (vihdzp) [inline Mathlib/Data/Sym/Sym2.lean:609]:
```suggestion
theorem isDiag_iff_equivSym_eq_replicate {s : Sym2 α} :
```

**Review by vihdzp — COMMENTED** [review]
some style stuff

**Comment by Jun2M** [comment]
Oh, no. what have I done.

**Pull request closed without merging**